src/rustdoc-json-types/lib.rs

```diff
@@ -52,11 +52,36 @@ pub struct Crate {
     pub paths: HashMap<Id, ItemSummary>,
     /// Maps `crate_id` of items to a crate name and html_root_url if it exists.
     pub external_crates: HashMap<u32, ExternalCrate>,
+    /// Information about the target for which this documentation was generated
+    pub target: Target,
     /// A single version number to be used in the future when making backwards incompatible changes
     /// to the JSON output.
     pub format_version: u32,
 }
 
+/// Information about a target
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub struct Target {
+    /// A list of `#[target_feature]` which exist for this target, along with their status in this
+    /// compiler session
+    pub target_features: Vec<TargetFeature>,
+}
+
+/// Information about a `#[target_feature]`
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub struct TargetFeature {
+    /// The name of this target feature
+    pub name: String,
+    /// Other target features which are implied by this target feature
+    pub implies_features: Vec<String>,
+    /// If this target feature is unstable, the name of the associated language feature gate
+    pub unstable: Option<String>,
```

Comment:
I don't love this name, `unstable` feels like it should be a bool. Maybe `feature_gate` is better, but open to something else.

Reply:
Hmm, I wanted to emphasize that the field here is describing the target feature's stability. How about `unstable_feature_gate`?

Reply:
Yeah, that works.